cleverhans/train.py

```diff
@@ -124,6 +129,23 @@ def train(sess, loss, x_train, y_train,
     with tf.control_dependencies(tf.get_collection(tf.GraphKeys.UPDATE_OPS)):
         train_step = optimizer.apply_gradients(grad)
 
+    if use_ema:
+        ema = tf.train.ExponentialMovingAverage(decay=ema_decay)
+        with tf.control_dependencies([train_step]):
+            train_step = ema.apply(var_list)
+        avg_params = [ema.average(param) for param in var_list]
+        tmp_params = [tf.Variable(param, trainable=False)
```

Comment:
Is the following used in place of a custom getter for tf.variable_scope ? Could you add a line or two explaining at a high level?

Reply:
I added a few lines of comments, let me know whether they make sense / whether they're sufficient